test/auth.ts

```diff
@@ -0,0 +1,123 @@
+import assert from 'assert'
+import { ImportMock } from 'ts-mock-imports'
+import * as configModule from '../lib/config'
+import { DropboxMiddleware } from '../lib/web/auth/dropbox'
+import { EmailMiddleware } from '../lib/web/auth/email'
+import { FacebookMiddleware } from '../lib/web/auth/facebook'
+import { GithubMiddleware } from '../lib/web/auth/github'
+import { GitlabMiddleware } from '../lib/web/auth/gitlab'
+import { GoogleMiddleware } from '../lib/web/auth/google'
+import { LdapMiddleware } from '../lib/web/auth/ldap'
+import { OAuth2Middleware } from '../lib/web/auth/oauth2'
+import { OPenIDMiddleware } from '../lib/web/auth/openid'
+import { TwitterMiddleware } from '../lib/web/auth/twitter'
+
+describe('AuthMiddlewares', function () {
+  // We currently exclude the SAML Auth, because it needs a certificate file
+  const middlewareList = [{
+    name: 'Facebook',
+    middleware: FacebookMiddleware,
+    config: {
+      facebook: {
+        clientID: 'foobar',
+        clientSecret: 'foobar'
+      }
+    }
+  }, {
+    name: 'Twitter',
+    middleware: TwitterMiddleware,
+    config: {
+      twitter: {
+        consumerKey: 'foobar',
+        consumerSecret: 'foobar'
+      }
+    }
+  }, {
+    name: 'GitHub',
+    middleware: GithubMiddleware,
+    config: {
+      github: {
+        clientID: 'foobar',
+        clientSecret: 'foobar'
+      }
+    }
+  }, {
+    name: 'Gitlab',
+    middleware: GitlabMiddleware,
+    config: {
+      gitlab: {
+        clientID: 'foobar',
+        clientSecret: 'foobar'
+      }
+    }
+  }, {
+    name: 'Dropbox',
+    middleware: DropboxMiddleware,
+    config: {
+      dropbox: {
+        clientID: 'foobar',
+        clientSecret: 'foobar'
+      }
+    }
+  }, {
+    name: 'Google',
+    middleware: GoogleMiddleware,
+    config: {
+      google: {
+        clientID: 'foobar',
+        clientSecret: 'foobar'
+      }
+    }
+  }, {
+    name: 'LDAP',
+    middleware: LdapMiddleware,
+    config: {
+      ldap: {
+        clientID: 'foobar',
+        clientSecret: 'foobar'
```

Comment:
ldap has a little bit different idea of how config should look like :D

Reply:
Well, actually it doesn't :laughing: 

As long as `config.ldap` is defined (but it can be an empty object), the `LDAPStrategy` constructor works and we don't test any "deeper" ^^